modules/ding_availability/js/ding_availability_labels.js

```diff
@@ -60,7 +60,13 @@
             // This can be triggered eventhough availability is not yet
             // fetched, which means it will show up as unavailable. So skip
             // items where we don't know if available or unavailable.
-            if (Drupal.DADB[entity_id] !== null) {
+            var allExists = true;
+            $.each(entity_ids, function(index, entity_id) {
+              if (Drupal.DADB[entity_id] === null) {
+                allExists = false;
+              }
+            });
+            if (allExists) {
               update_availability(id, entity_ids);
```

Comment:
Issues emerging here worry me a bit.

The current code might not be solid, but the fact that it requires changes may be a sign that we are missing something elsewhere in our integration. Based on the commit comment it is really hard to determine why this change is needed.

The approach in this PR jumps ship if there is a problem with any id. How about instead filtering `entity_ids` by removing entries that are not available as keys in `Drupal.DADB` and then passing the ones that are to `update_availability()`.

Reply:
This problem is caused by missing entity_id vs entity_ids, which causes issue as variable is undefined.

`update_availability` does a double loop to update availability. The way the AJAX work, you alway have all or nothing as everything is fetched in a single call, but the way the data structure is handled, detecting that in this context is not possible.
I opted for simple code that makes sense, so everything either works or fails, instead of having some weird hidden issues hard to detect, like missing availability for specific material types, which would be hard to catch, but missing availability for the entire material we can see and detect that we have an issue somewhere